kwasm-lemmas.md

```diff
@@ -335,6 +364,21 @@ They are non-trivial in their implementation, but the following should obviously
       [simplification]
 ```
 
+`#getRange` over `#setRange`
+
+```k
+    rule #getRange(#setRange(BM, EA, VALUE, SET_WIDTH), EA, GET_WIDTH)
+      => #wrap(GET_WIDTH *Int 8, VALUE)
+      requires         GET_WIDTH <=Int SET_WIDTH
+      [simplification]
+
+    rule #getRange(#setRange(BM, EA, VALUE, SET_WIDTH), EA, GET_WIDTH)
+      => #wrap(SET_WIDTH *Int 8, VALUE)
+         +Int (#getRange(BM, EA +Int SET_WIDTH, GET_WIDTH -Int SET_WIDTH) <<Int (SET_WIDTH *Int 8))
+      requires notBool GET_WIDTH <=Int SET_WIDTH
+      [simplificaton]
```

Comment:
I'm suspicious this lemma will help in general, since it makes the resulting term bigger.

Are you sure it's needed?

Reply:
I'll check. I think it's needed in some form at least, but perhaps the RHS of the addition resolves to 0 in that case, so this lemma can be specialized a bit. Trying it out.

Reply:
Okay, it worked. See the new lemmas.